### ISSUE TITLE
Improvements to teams models and endpoints

### ISSUE THREAD

**Issue by willemarcel** [hotosm/tasking-manager#2006]
Some questions and proposal changes related to teams:

On https://github.com/hotosm/tasking-manager/issues/1531 I saw that the approved proposal was not setting specific role to team members. But we yet have a `function` field on the `TeamMember` model.
* Should the organisation managers have management permissions on all its teams?
* Do we need a manager role to the `TeamMembers`? If yes, I would propose to have only two `TeamMember` roles: `MEMBER` and `MANAGER`. If no, let's remove the function field.

* We need an `active` field on `TeamMember`. It should be `false` if the user asked to join a team, but it needs approval from a manager and should be `true` after approval or when it doesn't need approval.

* The endpoint to allow the users to join a team is not working.

**Comment by xamanu** [comment]
* Yes, organisation admins shall be able to manage all it's teams.
* Yes we need two roles: `MEMBER` and something else. I suggest to use maybe `LEADER` just to avoid confusion between organization managers and program admins.
* Yes, an `active` field is needed.

**Comment by willemarcel** [comment]
Some details about how the join team endpoint should work:

URL: `/api/v2/teams/{team_id}/actions/join/`

payload: {
  "role": "string",
  "username": "string"
}

* The payload could be required only if the user is not a manager of the team organisation or leader of the team.
* On the case a manager/leader makes the request, the user is automatically added as a team member (`active` field should be `true`). On other cases, the active field should be `false`, as we yet need an approval to the user effectively join the team.
* If the request user doesn't fit the previous condition and is sending a payload, check if the username on the payload is equal to the request user, if not return 403.
* The role field on the payload should be checked only by the managers/leaders requests, on the other cases. the join role will be 'MEMBER'.
* We need another endpoint to approve the join requests (or simply use the same join endpoint updating the existing team member model).
* We need to add the `active` information to the members field on the teams endpoints.

**Comment by fitoria** [comment]
Quick question, what will happen when the user is not a a manager and requests to join a team, right now it just sends a notifications to managers but now it will just add is as inactive?

Would it be useful to add a "joined_date" on the team_members table to have information about that?

**Comment by willemarcel** [comment]
> Quick question, what will happen when the user is not a a manager and requests to join a team, right now it just sends a notifications to managers but now it will just add is as inactive?

We can configure notifications, but we will add a section on the teams detail page that will show the join requests, so the managers can approve / deny.

**Comment by xamanu** [comment]
> What will happen when the user is not a a manager and requests to join a team?

It depends what type of team. If the team is open to join, a user will just be added - and active.
If the team needs approval to join, we want to send notifications to the team leads, requesting their approval. And we want to be able to display the ones that are pending on the team page, so we need to flag them as `inactive`.

> Right now it just sends a notifications to managers but now it will just add is as inactive? Would it be useful to add a "joined_date" on the team_members table to have information about that?

Not sure about the `joined_date`. It seems like a nice to have. But also a bit of an overhead. How does this work with the approval? Is it the date a user asked to join? Or the date a user got approved? Then we might think about two columns with time stamps, but we don't really need it currently for the UI...
